src/compiler/emitter.ts

```diff
@@ -40,16 +40,18 @@ module ts {
         getSymbolAccessibilityDiagnostic: GetSymbolAccessibilityDiagnostic;
     }
 
-    interface AliasDeclarationEmitInfo {
-        declaration: ImportEqualsDeclaration;
+    interface ModuleElementDeclarationEmitInfo {
+        node: Node;
         outputPos: number;
         indent: number;
         asynchronousOutput?: string; // If the output for alias was written asynchronously, the corresponding output
+        subModuleElementDeclarationEmitInfo?: ModuleElementDeclarationEmitInfo[];
+        isVisible?: boolean;

```

Comment:
may be call this isReferenced or isUsed to distinguish it from the checker's isVisible flag..